src/api/baseapi.cpp

```diff
@@ -1276,7 +1276,12 @@ bool TessBaseAPI::ProcessPage(Pix *pix, int page_index, const char *filename,
 
   if (tesseract_->tessedit_write_images) {
     Pix *page_pix = GetThresholdedImage();
-    pixWrite("tessinput.tif", page_pix, IFF_TIFF_G4);
+    std::string output_filename = output_file_ + ".processed";
```

Comment:
This file is the result of thresholding, without other image preprocessing. 

I suggest this form:

`std::string output_filename = output_file_ + "-binary";`
or
`std::string output_filename = output_file_ + "-bw";`
or
`std::string output_filename = output_file_ + "-thresholded";`

I prefer `-` over `.`. IMO, the dot as file extension should only be used for standard extensions, not for custom, application-specific file extension.

Reply:
The current documentation says "Write processed input images to file". Can we be sure that there won't be other kinds of preprocessing in the future? You already added new thresholding methods.

Using `-` instead of `.` would be fine for me if others agree, too,